### PR TITLE
Fixed river generation

### PR DESCRIPTION
**Pull request by vader1986**
My last changes made the river generation slightly weird for water & mixed maps. The central point for map generation became a random point but the rivers are still generated at the middle of the map.

### CODE REVIEW

**Review by Flamefire — APPROVED** [review]
Thanks. Minor points I'll fix myself: MapPoint should be passed by copy, it is only 4 Bytes

**Review by Flamefire — APPROVED** [review]

File: libs/s25main/mapGenerator/RandomMap.h [hunk 1]
```diff
@@ -42,7 +42,7 @@ namespace rttr { namespace mapGenerator {
         Texturizer texturizer_;
         MapSettings settings_;
 
-        std::vector<River> CreateRivers();
+        std::vector<River> CreateRivers(const MapPoint& source = MapPoint::Invalid());
```
[the remark this answers is not in the document]
Comment (Flow86) [reply libs/s25main/mapGenerator/RandomMap.h]: I don't like this. Types you dont know how big they are should be always passed as reference.

[the remark this answers is not in the document]
Comment (Flamefire) [reply libs/s25main/mapGenerator/RandomMap.h]: There is a rule of thumb: Everything below the size of a Pointer should be passed by instance/copy. There are even templates to implement this in Boost.

Yours is correct in general, but for the Point/Extent types we know how big they are: They are all below the pointer size (neglecting 32 bit builds here, but MapPoint is 32 bit anyway)
Hence this is more efficient and done in almost all other places in the codebase (and I'm planning of making the remaining ones consistent too)

**Comment by vader1986** [comment]
Let me know if this need any more work!

**Comment by Flamefire** [comment]
Not from my side. I wanted to give Flow a chance to reply but i guess we can move that into a seperate issue if action is required. All good here

**Pull request merged**